Snakefile

```diff
@@ -0,0 +1,95 @@
+import glob
+import json
+import os
+#Funtion to extract samples from input JSON file and generate necessary .txt files
+N_FILES_MAX_PER_SAMPLE = config["n_files_max_per_sample"]
+def extract_samples_from_json(json_file):
+    output_files = []
+    with open(json_file, "r") as fd:
+        data = json.load(fd)
+        for sample, conditions in data.items():
+            for condition, details in conditions.items():
+                #Creating a filename for the file and condition
+                sample_name = f"{sample}__{condition}"
+                output_files.append((sample, condition))
+                # Write paths to a .txt file with the correct path replacement
+                with open(f"sample_{sample_name}_paths.txt", "w") as path_file:
+                    paths = [file_info["path"] for file_info in details["files"]]
+                    path_file.write("\n".join(paths))
+                    paths = [
+                        file_info["path"].replace(
+                            "https://xrootd-local.unl.edu:1094//store/user/AGC/nanoAOD",
+                            "root://eospublic.cern.ch//eos/opendata/cms/upload/agc/1.0.0/"
+                        )
+                        for file_info in details["files"]
+                    ]
+    return output_files
+#Function to get file paths based on the index
+def get_file_paths(wildcards, max=N_FILES_MAX_PER_SAMPLE):
+    "Return list of at most MAX file paths for the given SAMPLE and CONDITION."
+    filepaths = []
+    with open(f"sample_{wildcards.sample}__{wildcards.condition}_paths.txt", "r") as fd:
+        filepaths = fd.read().splitlines()
+    # Using the index as the wildcard, creating a path for each file based on it's index
+    return [f"histograms/histograms_{wildcards.sample}__{wildcards.condition}__{index}.root" for index in range(len(filepaths))][:max]
+
+samples_conditions = extract_samples_from_json("nanoaod_inputs.json")
+
+rule all:
+    input:
+        "hitograms.root"
+
+rule process_sample_one_file_in_sample:
+    container:
+        "docker.io/reanahub/reana-demo-agc-cms-ttbar-coffea:1.0.0"
+    resources:
+        kubernetes_memory_limit="1850Mi"
+    input:
+        notebook="ttbar_analysis_reana.ipynb"
+    output:
+        "histograms/histograms_{sample}__{condition}__{index}.root"
+    params:
+        sample_name = "{sample}__{condition}"
+    shell:
+        "/bin/bash -l && source fix-env.sh && "
+        "python prepare_workspace.py sample_{params.sample_name}_{wildcards.index} > directory_name.txt && "
```

Comment:
I'll change this to include `prepare_workspace.py` call directly to the second command in order to avoid using a temporary file name (that may perhaps clash if this rule is scattered many times across input).